src/portal-impl.c

```diff
@@ -198,18 +309,163 @@ load_installed_portals (gboolean opt_verbose)
   implementations = g_list_sort (implementations, sort_impl_by_use_in_and_name);
 }
 
+static PortalConfig *
+load_portal_configuration_for_dir (gboolean opt_verbose, const char *base_directory, const char *portal_file)
+{
+  g_autofree char *path = g_build_filename (base_directory, portal_file, NULL);
+  g_autoptr(GKeyFile) key_file = g_key_file_new ();
+
+  g_debug ("Looking for portals configuration in '%s'", path);
+  if (!g_key_file_load_from_file (key_file, path, G_KEY_FILE_NONE, NULL))
+    return NULL;
+
+  g_auto(GStrv) ifaces = g_key_file_get_keys (key_file, "preferred", NULL, NULL);
+
+  if (ifaces != NULL)
+    {
+      g_autoptr(GPtrArray) interfaces = g_ptr_array_new_full (g_strv_length (ifaces) + 1, NULL);
+      g_autoptr(PortalConfig) conf = g_new0 (PortalConfig, 1);
+
+      for (size_t i = 0; ifaces[i] != NULL; i++)
+        {
+          g_autoptr(PortalInterface) interface = g_new0 (PortalInterface, 1);
+
+          /* Special case "default" as the fallback */
+          if (strcmp (ifaces[i], "default") == 0)
+            interface->dbus_name = g_strdup ("default");
+          else
+            interface->dbus_name = g_strdup (ifaces[i]);
```

Comment:
Aren't both branches the same?

Reply:
Yes, though it makes it clear which one it is; I'm also open to changing the default key.